airflow/jobs/scheduler_job.py

```diff
@@ -872,7 +876,7 @@ def _change_state_for_tis_without_dagrun(
                     models.TaskInstance.execution_date ==
                     subq.c.execution_date) \
                 .update({models.TaskInstance.state: new_state}, synchronize_session=False)
```

Comment:
`ti.set_state` also sets end_time, and possibly duratiuon -- this was probably missing here! (Can be a separate PR)

Reply:
Will do it in a separate PR